engine/app/components/citizens_advice_components/on_this_page.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module CitizensAdviceComponents
+  class OnThisPage < Base
+    attr_accessor :headings
+
+    renders_many :links, "Link"
+
+    def initialize(show_nested_links: false)
+      super
+      @show_nested_links = show_nested_links
+    end
+
+    def render?
+      links.present?
+    end
+
+    def modifier_class_names
+      if show_nested_links? && children_exist?
+        "js-cads-on-this-page cads-on-this-page--expandable"
+      elsif columns?
+        "cads-on-this-page--columns"
+      end
+    end
+
+    def show_children_for?(link)
+      show_nested_links? && link.children.present?
+    end
+
+    private
+
+    def columns?
+      links.size > 3
+    end
+
+    def show_nested_links?
+      @show_nested_links
+    end
+
+    def children_exist?
```

Comment:
I'm not sure I've done the children_exist logic properly so it's worth checking

Reply:
The logic looks OK to me.

You can simplify it a bit by not passing the link into its own `children?` method, and use the instance variable `@children` to see if they are present.

So you would have
```rb 
def children_exist?
  links.any? { |link| link.children? }
end
```
and in `class Link` have
```rb
def children?
   @children.present?
end
```